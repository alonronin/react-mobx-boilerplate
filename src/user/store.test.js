import Store from './store';

describe('User store', () => {
	test('Store snap', () => {
		const store = new Store();

		expect(store).toMatchSnapshot();
	});

	test('Store object', () => {
		const store = new Store();
		const first = store.first;
		const last = store.last;
		const fullName = store.fullName;

		expect({
			first,
			last,
			fullName
		}).toMatchObject({
			first: 'Alon',
			last: 'Valadji',
			fullName: 'Alon Valadji'
		});
	});

	test('Store setState', () => {
		const store = new Store();
		store.setState({ first: 'Test', last: 'Test'});

		expect(store).toMatchSnapshot();
	})
});

