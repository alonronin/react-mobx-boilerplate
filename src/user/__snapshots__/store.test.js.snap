// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User store Store setState 1`] = `
Object {
  "first": "Test",
  "fullName": "Test Test",
  "last": "Test",
}
`;

exports[`User store Store snap 1`] = `
Object {
  "first": "Alon",
  "fullName": "Alon Valadji",
  "last": "Valadji",
}
`;
